### ISSUE TITLE
Wrong template is used when a custom ContentController is used

### ISSUE THREAD

**Issue by maximilian-walter** [FluidTYPO3/flux#1691]
I have updated Flux from 9.1.0 to. 9.2.0 today and now I'm getting the following Exception:

> #1257246929 TYPO3Fluid\Fluid\View\Exception\InvalidTemplateResourceException
Tried resolving a template file for controller action "Content->recordlist" in format ".html", but none of the paths contained the expected template file (Content/Recordlist.html). The following paths were checked: /public/typo3conf/ext/example_site/Resources/Private/Templates/

The following template causes this Exception:
`/public/typo3conf/ext/example_site/Resources/Private/Templates/Content/RecordList.html`.

-> The problem is the generated filename: It should be `RecordList.html`, but Fluid uses`"Recordlist.html` (with a lowercase "L").

My templates are registered in the file `ext_localconf.php` with the following call:

```php
`\FluidTYPO3\Flux\Core::registerProviderExtensionKey('MyVendor.ExampleSite', 'Content');`.
```

I'm using a custom ContentController `\MyVendor\ExampleSite\Controller\ContentController` which contains a action for this template:

```php
<?php
declare(strict_types=1);
namespace MyVendor\ExampleSite\Controller;

use FluidTYPO3\Flux\Controller\AbstractFluxController;

class ContentController extends AbstractFluxController
{
    public function recordListAction(): void
    {
        $this->view->assign('records', $this->loadRecords($this->getData()));
    }

    protected function loadRecords(array $data): array
    {
        // Load records from database
        return $loadedData;
    }
}
```

When the action is removed everything works as expected (but the template has no data...).

-> Flux behaves differently when the ContentController has a action for a template. I tried to find the change between 9.1.0 and 9.2.0 which causes this bug and I gues the following commit introduced this regression:
https://github.com/FluidTYPO3/flux/commit/c6c17094ec932a93576406a2d84f2f60cd3561c9

The method `\FluidTYPO3\Flux\Integration\ContentTypeBuilder::configureContentTypeFromTemplateFile()` has a variable $emulatedControllerAction which contains the wrong value ("Recordlist" instead of "RecordList") when $contentType is not null.
https://github.com/FluidTYPO3/flux/blob/c6c17094ec932a93576406a2d84f2f60cd3561c9/Classes/Integration/ContentTypeBuilder.php#L60

I have no idea why this change was made and how to fix it without breaking everything else, so I'm not able to provide a patch. But I hope my explanations are enough to reproduce this issue. :-)

Thanks for your hard work! I'm using Flux for several years now and it served me very well.

**Comment by NamelessCoder** [comment]
Hi @maximilian-walter,

I can see how this would happen. The expected controller action name is resolved based on the $contentType that gets passed, by selecting what's after the first `_`. But the $contentType in your case is `xyz_recordlist` because the content type name is created by forcing `strtolower` and replacing underscores, then prefixing that with the extension signature.

The template filename no longer applies if a $contentType is passed and Flux will do that now. The reason is that a new feature allows the same template file to integrate as multiple different CType values so using the template filename as the only way to determine the controller action is no longer feasible.

I'll think about how to solve this but in the meantime perhaps you can find a way to make $contentType null in your case. You can also work around this by registering the individual template files in which case you can leave out the $contentType or make it null so the filename gets used as action name, preserving the `lowerCamelCase` format. Or perhaps Flux's feature to scan for template files and register them, should never attempt to set a CType for *those* particular files so we fall back to the old behavior for auto-registered files.

**Comment by NamelessCoder** [comment]
More specifically:

https://github.com/FluidTYPO3/flux/commit/c6c17094ec932a93576406a2d84f2f60cd3561c9#diff-258571a06bf50ff08d5517a56925d952R91

This is where the $contentType gets a value if it is null, which is preventing templates from having null as desired CType, which then triggers the unexpected conversion in the line you identified.

**Comment by NamelessCoder** [comment]
A solution be that `static::determineContentType($providerExtensionName, $templateFilename);` happens in https://github.com/FluidTYPO3/flux/commit/c6c17094ec932a93576406a2d84f2f60cd3561c9#diff-15cdc00e2d88ebbd6728d836ff9e11c4R66 instead of https://github.com/FluidTYPO3/flux/commit/c6c17094ec932a93576406a2d84f2f60cd3561c9#diff-258571a06bf50ff08d5517a56925d952R91. This would keep the null value until the very last moment and only use CType if you definitely specified it yourself (which the new features that require it, will do).

**Comment by moe2k** [comment]
I think I ran into the same or a similar issue, using TYPO3 9 LTS, ddev and flux with custom ContentController.

My flux form id was "tableOfContents", the template file "TableOfContents.html". 

The behavior was totally strange. Changes to the Template's Main Section were just never applied. Debug output was still there even though I removed the <f:debug> vh. Breaking the flux form section (by using non-existing vh) successfully broke the rendering, though. But after fixing still no effect. 

It seems like it was just hard-cached somewhere, I just don't know where.

Even though I was using ext:uncache, delete cache through backend, manually deleting all caches (rm -Rf /var/cache, rm -Rf typo3temp/var, all database tables with cache*, typo3cms cache:flushcomplete - all inside my ddev webserver container). I also grepped the whole ddev webserver container filesystem for the deleted string. 

After renaming id to "tableofcontents" and template to "Tableofcontents.html" the issue was gone. Renaming back to the old camelCase schema brings the issue back. Maybe we should point out somewhere that IDs have to be lowercase and Templates should use the "first letter uppercase" convention?
